pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/worker/WorkerConfig.java

```diff
@@ -738,6 +738,17 @@ public String getFunctionAuthProviderClassName() {
     )
     private List<String> additionalJavaRuntimeArguments = new ArrayList<>();
 
+    @FieldContext(
+            category = CATEGORY_CONNECTORS,
+            doc = "Whether to ignore unknown properties when deserializing the connector configuration. "
+                    + "After upgrading a connector to a new version with a new configuration, "
+                    + "the new configuration may not be compatible with the old connector. "
+                    + "In case of rollback, it's required to also rollback the connector configuration. "
+                    + "Ignoring unknown fields makes possible to keep the new configuration and "
+                    + "only rollback the connector."
+    )
+    private boolean ignoreUnknownConfigFields = false;
```

Comment:
This configuration applies to the instances of the functions/connectors, and not to the function worker (that already  ignores unknown fields)

what about 'functionsIgnoreUnknownConfigFields' ?    (maybe we can do better, but connectors are actually functions)

Reply:
I thought about that and I agree. But there are other properties that are meant to be used by the java runner like 
maxPendingAsyncRequests, exposeAdminClientEnabled, additionalJavaRuntimeArguments  so I'd prefer to stick with this convention